File: CS/WindowsFormsApplication40/MyGalleryControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using System.Windows.Forms;
using System.Drawing;

namespace WindowsFormsApplication40
{
    class MyGalleryControl : GalleryControl
    {
        protected override void OnKeyDown(System.Windows.Forms.KeyEventArgs e)
        {
            base.OnKeyDown(e);
            if (KeyboardSelectedItemChanged != null)
                if (e.KeyCode == Keys.Left ||
                    e.KeyCode == Keys.Right ||
                    e.KeyCode == Keys.Up||
                    e.KeyCode ==Keys.Down)
            { StandaloneGalleryViewInfo View =(this.Gallery as MyGalleryControlGallery).MyViewInfo;
            KeyboardSelectionEventArgs args = new KeyboardSelectionEventArgs(
                View.KeyboardSelectedItemIndex,
                View.KeyboardSelectedItem,
                View.KeyboardSelectedGroupIndex,
                View.KeyboardSelectedGroup);
                KeyboardSelectedItemChanged(this, args);
            }
        }
        protected override DevExpress.XtraBars.Ribbon.Gallery.GalleryControlGallery CreateGallery()
        {
            return new MyGalleryControlGallery(this);
        }
        public delegate void KeyboardSelectedItemChangedHandler(Object sender,KeyboardSelectionEventArgs e);
        
        public event KeyboardSelectedItemChangedHandler KeyboardSelectedItemChanged;

    }
}

File: VB/WindowsFormsApplication40/MyGalleryControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports System.Windows.Forms
Imports System.Drawing

Namespace WindowsFormsApplication40
	Friend Class MyGalleryControl
		Inherits GalleryControl
		Protected Overrides Sub OnKeyDown(ByVal e As System.Windows.Forms.KeyEventArgs)
			MyBase.OnKeyDown(e)
			If KeyboardSelectedItemChangedEvent IsNot Nothing Then
				If e.KeyCode = Keys.Left OrElse e.KeyCode = Keys.Right OrElse e.KeyCode = Keys.Up OrElse e.KeyCode =Keys.Down Then
				Dim View As StandaloneGalleryViewInfo =(TryCast(Me.Gallery, MyGalleryControlGallery)).MyViewInfo
			Dim args As New KeyboardSelectionEventArgs(View.KeyboardSelectedItemIndex, View.KeyboardSelectedItem, View.KeyboardSelectedGroupIndex, View.KeyboardSelectedGroup)
				RaiseEvent KeyboardSelectedItemChanged(Me, args)
				End If
			End If
		End Sub
		Protected Overrides Function CreateGallery() As DevExpress.XtraBars.Ribbon.Gallery.GalleryControlGallery
			Return New MyGalleryControlGallery(Me)
		End Function
		Public Delegate Sub KeyboardSelectedItemChangedHandler(ByVal sender As Object, ByVal e As KeyboardSelectionEventArgs)

		Public Event KeyboardSelectedItemChanged As KeyboardSelectedItemChangedHandler

	End Class
End Namespace
